src/script/api/script_text.cpp

```diff
@@ -150,13 +148,34 @@ SQInteger ScriptText::_set(HSQUIRRELVM vm)
 		return SQ_ERROR;
 	}
 
-	if (k > SCRIPT_TEXT_MAX_PARAMETERS) return SQ_ERROR;
 	if (k < 1) return SQ_ERROR;
 	k--;
 
 	return this->_SetParam(k, vm);
 }
 
+/**
+ * Set the number of padding parameters to use, for compatibility with old scripts.
+ * This is called during RegisterGameTranslation.
+ */
+void ScriptText::SetPadParameterCount(HSQUIRRELVM vm)
+{
+	ScriptText::pad_parameter_count = 0;
+
+	sq_pushroottable(vm);
+	sq_pushstring(vm, "GSText", -1);
+	if (!SQ_FAILED(sq_get(vm, -2))) {
+		sq_pushstring(vm, "SCRIPT_TEXT_MAX_PARAMETERS", -1);
+		if (!SQ_FAILED(sq_get(vm, -2))) {
+			SQInteger value;
+			if (!SQ_FAILED(sq_getinteger(vm, -1, &value))) {
+				ScriptText::pad_parameter_count = value;
+			}
+		}
+	}
+	sq_pop(vm, 3);
```

Comment:
If it fails to retrieve `GSText` class, then only 2 values are pushed.
```suggestion
	SQInteger top = sq_gettop(vm);
	sq_pushroottable(vm);
	sq_pushstring(vm, "GSText", -1);
	if (!SQ_FAILED(sq_get(vm, -2))) {
		sq_pushstring(vm, "SCRIPT_TEXT_MAX_PARAMETERS", -1);
		if (!SQ_FAILED(sq_get(vm, -2))) {
			SQInteger value;
			if (!SQ_FAILED(sq_getinteger(vm, -1, &value))) {
				ScriptText::pad_parameter_count = value;
			}
		}
	}
	sq_settop(vm, top);
```